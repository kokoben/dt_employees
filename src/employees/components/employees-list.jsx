import React, { Component } from 'react';
import { List } from 'antd';

const EmployeesList = props => (
  <div>
  </div>
);

export default EmployeesList;

