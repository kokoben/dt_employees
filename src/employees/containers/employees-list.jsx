import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { List } from 'antd';
import { setEmployees } from '../actions';

class EmployeesList extends Component {
  componentDidMount() {
    this.props.setEmployees(1, 100);
  }

  render() {
    return (
      <div>
        poop list
      </div>
    );
  }
}

EmployeesList.propTypes = {
  setEmployees: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  employees: state.employees.employees,
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setEmployees,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(EmployeesList);

