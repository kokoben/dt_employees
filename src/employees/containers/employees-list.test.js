import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { EmployeesList } from './employees-list';

it('renders correctly', () => {
  const mockSetEmployees = jest.fn();
  const mockSetCursor = jest.fn();
  const mockSetCurrentPage = jest.fn();

  const wrapper = shallow(<EmployeesList
    setEmployees={mockSetEmployees}
    setCursor={mockSetCursor}
    setCurrentPage={mockSetCurrentPage}
    cursor={0}
    currentPage={1}
  />);

  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
