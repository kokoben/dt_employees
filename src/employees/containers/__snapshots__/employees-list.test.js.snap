// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    City of Chicago Employee Directory
  </h1>
  <Spin
    prefixCls="ant-spin"
    size="large"
    spinning={true}
    style={
      Object {
        "marginTop": "20px",
      }
    }
    tip="Loading employees..."
    wrapperClassName=""
  />
</div>
`;
